pyspark_datasources/kaggle.py

```diff
@@ -0,0 +1,108 @@
+import tempfile
+from functools import cached_property
+from typing import Iterator
+
+import pyarrow as pa
```

Comment:
Do we have to depend on pyarrow? Can we throw a better error message if pyarrow is not installed?

Reply:
Good point. Depending on pyarrow should be fine since pyspark data source itself depends on pyarrow. Let's import pyarrow later so that pyspark shows the error message if it's missing.